app/blueprints/fund_builder/routes.py

```diff
@@ -204,80 +208,208 @@ def clone_round(round_id, fund_id):
-                opens=get_datetime(form.opens),
-                deadline=get_datetime(form.deadline),
-                assessment_start=get_datetime(form.assessment_start),
-                reminder_date=get_datetime(form.reminder_date),
-                assessment_deadline=get_datetime(form.assessment_deadline),
-                prospectus_link=form.prospectus_link.data,
-                privacy_notice_link=form.privacy_notice_link.data,
-                contact_us_banner_json={"en": form.contact_us_banner_json.data, "cy": None},
-                reference_contact_page_over_email=form.reference_contact_page_over_email.data == "true",
-                contact_email=form.contact_email.data,
-                contact_phone=form.contact_phone.data,
-                contact_textphone=form.contact_textphone.data,
-                support_times=form.support_times.data,
-                support_days=form.support_days.data,
-                instructions_json={"en": form.instructions_json.data, "cy": None},
-                feedback_link=form.feedback_link.data,
-                project_name_field_id=form.project_name_field_id.data,
-                application_guidance_json={"en": form.application_guidance_json.data, "cy": None},
-                guidance_url=form.guidance_url.data,
-                all_uploaded_documents_section_available=form.all_uploaded_documents_section_available.data == "true",
-                application_fields_download_available=form.application_fields_download_available.data == "true",
-                display_logo_on_pdf_exports=form.display_logo_on_pdf_exports.data == "true",
-                mark_as_complete_enabled=form.mark_as_complete_enabled.data == "true",
-                is_expression_of_interest=form.is_expression_of_interest.data == "true",
-                feedback_survey_config=form.feedback_survey_config.data,
-                eoi_decision_schema=form.eoi_decision_schema.data,
-            )
-        )
+    form = RoundForm()
 
-        flash(f"Saved round {form.title_en.data}")
-        return redirect(url_for("build_fund_bp.index"))
+    if round_id:
+        round = get_round_by_id(round_id)
+        form = populate_form_with_round_data(round)
+
+    if form.validate_on_submit():
+        if round_id:
+            update_existing_round(round, form)
+            flash(f"Updated round {round.title_json['en']}")
+        else:
+            create_new_round(form)
+            flash(f"Created round {form.title_en.data}")
+        return redirect(url_for(BUILD_FUND_BP_INDEX))
 
     return render_template(
         "round.html",
         form=form,
         all_funds=all_funds_as_govuk_select_items(all_funds),
+        round_id=round_id,
+    )
+
+
+def populate_form_with_round_data(round):
+    """
+    Populate a RoundForm with data from a Round object.
+
+    :param Round round: The round object to populate the form with
+    :return: A RoundForm populated with the round data
+    """
+    round_data = {
```

Comment:
Cant we do this mapping logic inside the RoundForm it self ? Also In java we can use ObjectMappers what it does is if an object has some variable it automatically maps those so we can reduce number of code that we have and & it is efficient (Again a observation)